pkg/clustermanager/cluster_manager.go

```diff
@@ -395,6 +396,14 @@ func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster,
 		return fmt.Errorf("waiting for workload cluster control plane to be ready: %v", err)
 	}
 
+	if provider.Name() == constants.CloudStackProviderName {
```

Comment:
Can we introduce a new method in the provider or add it to an existing one, or is that not possible?

Reply:
Can we not add this logic to here? https://github.com/aws/eks-anywhere/blob/main/pkg/clustermanager/cluster_manager.go#L376

Reply:
See https://github.com/aws/eks-anywhere/issues/2061 for context about the long-term home for this operation and why we can't use the existing RunPostControlPlaneUpgrade method